### ISSUE TITLE
Selected area gets reduced after "replace all in selection"

### ISSUE THREAD

**Issue by fml2** [notepad-plus-plus/notepad-plus-plus#2995]
This happens with NP++ v.7.3.1 32 bit.

### Description of the Issue
After performing "replace all" action with "in selection" check box checked, the selected area is reduced (the right most character is not selected anymore).

### Steps to Reproduce the Issue

1. Create a new document
2. type "aaaaa" (five times "a")
3. select the middle three a's
4. press ctrl+h (replace dialog), replace "a" with "b", check "in selection"
5. press "Replace all"

Result: the selected a's are replaced by b's (that's correct) but only two b's are selected. The rightmost "b" is not selected.

### Expected Behavior
I'd expect the selection to be on all three b's in this case.

**Comment by sasumner** [comment]
I found a similar but slightly different scenario:

1. create several lines of "aaa"
2. select all the lines but do not include the line ending in the selection on the last line
3. replace-all in-selection with "bbb"
4. note that after the operation the last line has been changed but is no longer included in the selection

In step 2, if you DO include the line ending in the selection, after step 3 is performed the selection is correct (hasn't changed).

I don't know if this addition has any value, but I found interesting that it acts differently if the line-ending is included versus not included.
